src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,114 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { client } from '../utils/fetchClient';
+
+type Props = {
+  selectedPost: Post | null,
+  setPostsComments: React.Dispatch<React.SetStateAction<Comment[]>>,
+  setCommentErrorMessage: (value: string) => void,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  setPostsComments,
+  setCommentErrorMessage,
+}) => {
+  const [isFormLoading, setIsFormLoading] = useState(false);
+  const [commentName, setCommentName] = useState('');
+  const [commentEmail, setCommentEmail] = useState('');
+  const [commentText, setCommentText] = useState('');
+  const [formError, setFormError] = useState({
+    name: '',
+    email: '',
+    text: '',
+  });
+
+  const isEmptyField = !!commentName.length
+    || !commentEmail.length
+    || !commentText.length;
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCommentName(event.target.value);
+    setFormError({ ...formError, name: '' });
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCommentEmail(event.target.value);
+    setFormError({ ...formError, email: '' });
+  };
+
+  const handleTextChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setCommentText(event.target.value);
+    setFormError({ ...formError, text: '' });
+  };
+
+  const createComment = ({
+    postId,
+    name,
+    email,
+    body,
+  }: Omit<Comment, 'id'>) => {
+    return client.post<Comment>('/comments', {
+      postId, name, email, body,
+    });
+  };
+
+  const addComment = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!commentName) {
+      setFormError(rest => ({ ...rest, name: 'Name is required' }));
+    }
+
+    if (!commentEmail) {
+      setFormError(rest => ({ ...rest, name: 'Email is required' }));
+    }
+
+    if (!commentText) {
+      setFormError(rest => ({ ...rest, name: 'Enter some text' }));
+    }
+
+    if (selectedPost?.id && !isEmptyField) {
```

Comment:
in case `selectedPost?.id` is 0 number it will not work

generally, it doesn't work, also, there is a strange error for the name field:
[Uploading Screencast from 28-09-23 19:49:44.webm…]()

Reply:
Not fixed